ir/expression.cpp

```diff
@@ -137,3 +137,45 @@ IR::Constant IR::Constant::operator-() const { return IR::Constant(-value); }
 IR::Constant IR::Constant::GetMask(unsigned width) {
     return (IR::Constant(1) << width) - IR::Constant(1);
 }
+
+const IR::Constant *IR::Constant::get(const IR::Type *t, big_int v, Util::SourceInfo si) {
+    // Only cache bits with width lower than 16 bit to restrict the size of the cache.
+    const auto *tb = t->to<Type_Bits>();
+    if (t->width_bits() > 16 || tb == nullptr) {
+        return new IR::Constant(si, t, v);
+    }
+    // Constants are interned. Keys in the intern map are pairs of types and values.
+    using key_t = std::tuple<int, RTTI::TypeId, bool, big_int>;
+    static std::map<key_t, const Constant *> CONSTANTS;
```

Comment:
suggestion: Use an abseil map with the new hasher?

Reply:
`big_int` does not have an implementation for hashing it looks like? It works for the string literals at least.